Recover help text into new forms -  issue328.

kate-mills opened the pull request:
You can check it out here: https://issue328-dot-rebuildingtogethercaptain-hrd.appspot.com

Is the **verbose_name** attribute in old models.py what the user sees/label?
If so, I am confused by:
	  phone1.verbose_name = 'Preferred Phone (deprecated)'
	  phone2 = db.PhoneNumberProperty()
	  phone2.verbose_name = 'Backup Phone (deprecated)'

kate-mills committed:
updated captain with help_text into new forms

kate-mills commented on app/js/models/captain.js:
```diff
@@ -4,8 +4,16 @@ define(
+                if(!attrs.name){
+                    this.errorModel.set({name: "Please enter captain name."})
+                }
+                if (!attrs.email){
```
Luke, Even though email is not required in ndb_models, I get a server error if email not entered into form.
Is this the best way to handle this?

babybunny replied:
Yes, it's best to catch any errors within the browser layer, if you can.  Because you can give quick feedback to the user. 

The email field should not be required.  There are some captains who don't use email.  We should fix the server error.  Can you file a separate issue please?

kate-mills commented on app/js/views/captain.js:
```diff
@@ -12,14 +12,6 @@ define(
                 disabled: true
             },
             {
-                // Unsure if user wants to view last_editor here?
```
I found the following in old forms.py, and I assume the property should not be included here. 

class CaptainForm(djangoforms.ModelForm):
	  class Meta:
	    model = models.Captain
	    exclude = ['modified', '**last_editor**', 'last_welcome', 'search_prefixes']

babybunny replied:
Correct, we should omit 'last_editor' from forms.

kate-mills committed:
Added help text/ validation to checkrequest form

babybunny commented:
Yeah the verbose_name field was a way to override the automatic name.  There's also a help_text field in the old models.py which was used as a more detailed description including examples.

kate-mills committed:
updated captain form.  email is not required

kate-mills committed:
updated inkinddonation form with help text

kate-mills committed:
added help text and validation to item form

kate-mills committed:
added validation to order form.

babybunny commented:
Is this ready to merge?  Any work ongoing here?

kate-mills commented:
No, this is not ready to merge.  I am still working on it.

On Fri, Sep 22, 2017 at 10:02 AM, Luke Stone <notifications@github.com>
wrote:

> Is this ready to merge? Any work ongoing here?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/babybunny/rebuildingtogethercaptain/pull/336#issuecomment-331503727>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/APciz8jiGIjChjjCEIYy2JTtZhuiypglks5sk-gxgaJpZM4PZeqt>
> .
>

kate-mills committed:
updated help text and  validation to ordersheet form

kate-mills committed:
updated help text and validation to site form

kate-mills committed:
updated help text and validation to stafftime form

kate-mills committed:
updated help text and validation to vendorrecipt form

kate-mills committed:
removed all validation changes added to this issue - now changes/issue match

kate-mills committed:
removed period in label

kate-mills commented:
Finally finished!  Please review when you get a moment.
Check it out here: [https://issue328-dot-rebuildingtogethercaptain-hrd.appspot.com/room/staff_home](https://issue328-dot-rebuildingtogethercaptain-hrd.appspot.com/room/staff_home)

babybunny commented:
Ok will do.

On Sep 24, 2017 2:38 PM, "Kate Mills" <notifications@github.com> wrote:

> Finally finished! Please review when you get a moment.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/babybunny/rebuildingtogethercaptain/pull/336#issuecomment-331742029>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AK-yPFva8sBj5rW529Cvgjjfb6BW-bdLks5slsvWgaJpZM4PZeqt>
> .
>

babybunny commented on app/js/views/order_flow.js:
```diff
@@ -66,8 +66,15 @@ define(
+            //     name: "notes",
+            //     label: "Instructions for warehouse staff",
+            //     control: "textarea"
+            // }
```
you can remove this.  it's not much use to have commented-out code. we have all the change history so we can always get code back if we need to restore something.  basically people don't really like to see commented-out code in commits (though it's fine for temporary things while you're working).

babybunny commented on app/js/views/vendorreceipt.js:
```diff
@@ -70,9 +69,11 @@ define(
-                    room_model_module: SupplierChoice
+                    room_model_module: SupplierChoice,
+                    // helpMessage found -> old forms.py line 257
+                    // helpMessage: "or add a new vendor using the form on the right"
```
I created a new issue to track this
https://github.com/babybunny/rebuildingtogethercaptain/issues/346

so you can remove the comment.  there is another similar case in stafftime.js.

babybunny commented on app/js/views/checkrequest.js:
```diff
@@ -38,7 +64,7 @@ define(
                 label: "Description",
                 control: "textarea",
-                helpMessage: "Please include place of purchase and list of items purchased.  Also submit corresponding recipt."
+                helpMessage: "Please include place of purchase and list of items purchased<br>and submit corresponding recipt."
```
If we remove the <br>, does something bad happen?  I can't see wht would go wrong.

Usually we are trying to separate the information (like the sentence) from the display (formatting details like the <br>).  So I kind of cringe when I see HTML stuck in the middle of JavaScript strings.  But this is so minor that I don't really mind.

kate-mills replied:
No cringing!! Not acceptable!
I  am not sure what comma you are referring to.
I have the sentence wrong anyhow, it should have parentheses inside the quotes.
I'll take another look at this.

kate-mills replied:
Do you want this label separated into two lines?  
Is that why you have it broken up into two strings?  

label=('Description '
             '(Please include place of purchase and list of items purchased '
             'and submit corresponding receipt.)')
  )

I am not sure how to do it *yet* without \<br\>.

This is what I have tried: 

**escaping**
'first     **\n**     string'   =   *string on one line*
'first     **\r**     string'   =   *string on one line*

**concat**
'first '   **+**   'second '  =   *string on one line*
'first '.concat('second')   =   *string on one line*

**comma**
('first '   **,**   'second '  )  =    *error*

kate-mills replied:
I will try the comma.  I think it looks better.

babybunny reviewed (changes requested):
Looks really good!

I have a few specific comments for you to address.  Should be easy.

kate-mills committed:
removed unnecessary commeents and commented-out code

kate-mills committed:
removed span from js

kate-mills committed:
replaced <br> with comma

kate-mills committed:
removed <br> to keep help text consistent

kate-mills commented:
I removed the commented-out code and some of my unnecessary comments within the code.
Also, I removed the html within the js.
It looks better anyway.
I deployed these updates to [https://issue328-dot-rebuildingtogethercaptain-hrd.appspot.com/room/staff_home](https://issue328-dot-rebuildingtogethercaptain-hrd.appspot.com/room/staff_home)

babybunny commented:
Nice.  I think there is a button for request review.  Something to
explicitly alert me that you are done and ready


On Sep 25, 2017 6:55 PM, "Kate Mills" <notifications@github.com> wrote:

I removed the commented-out code and some of my unnecessary comments within
the code.
Also, I removed the html within the js.
It looks better anyway.
I deployed these updates to https://issue328-dot-
rebuildingtogethercaptain-hrd.appspot.com/room/staff_home

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/babybunny/rebuildingtogethercaptain/pull/336#issuecomment-332063344>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/AK-yPPiz2oQscGVupYWWOdesAxMA-XAjks5smFmtgaJpZM4PZeqt>
.

kate-mills commented:
I found the button!  Thanks!

babybunny commented:
Yes that's it!!!  Will look tonight

On Sep 26, 2017 3:02 PM, "Kate Mills" <notifications@github.com> wrote:

> I found the button! Thanks!
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/babybunny/rebuildingtogethercaptain/pull/336#issuecomment-332349380>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AK-yPLZdzfAvnrwuspqEya0rnRW_lLBwks5smXRzgaJpZM4PZeqt>
> .
>

babybunny commented:
I approved it!  you can go ahead and merge into 'service'

On Tue, Sep 26, 2017 at 9:09 PM, Luke Stone <babybunny@gmail.com> wrote:

> Yes that's it!!!  Will look tonight
>
> On Sep 26, 2017 3:02 PM, "Kate Mills" <notifications@github.com> wrote:
>
>> I found the button! Thanks!
>>
>> —
>> You are receiving this because your review was requested.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/babybunny/rebuildingtogethercaptain/pull/336#issuecomment-332349380>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AK-yPLZdzfAvnrwuspqEya0rnRW_lLBwks5smXRzgaJpZM4PZeqt>
>> .
>>
>

kate-mills commented:
Yay!! Thank you!  

Sent from my iPhone

> On Sep 26, 2017, at 10:44 PM, Luke Stone <notifications@github.com> wrote:
> 
> I approved it! you can go ahead and merge into 'service'
> 
> On Tue, Sep 26, 2017 at 9:09 PM, Luke Stone <babybunny@gmail.com> wrote:
> 
> > Yes that's it!!! Will look tonight
> >
> > On Sep 26, 2017 3:02 PM, "Kate Mills" <notifications@github.com> wrote:
> >
> >> I found the button! Thanks!
> >>
> >> —
> >> You are receiving this because your review was requested.
> >> Reply to this email directly, view it on GitHub
> >> <https://github.com/babybunny/rebuildingtogethercaptain/pull/336#issuecomment-332349380>,
> >> or mute the thread
> >> <https://github.com/notifications/unsubscribe-auth/AK-yPLZdzfAvnrwuspqEya0rnRW_lLBwks5smXRzgaJpZM4PZeqt>
> >> .
> >>
> >
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

babybunny commented:
woot!

On Wed, Sep 27, 2017 at 2:46 PM, Kate Mills <notifications@github.com>
wrote:

> Merged #336
> <https://github.com/babybunny/rebuildingtogethercaptain/pull/336>.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/babybunny/rebuildingtogethercaptain/pull/336#event-1268691685>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AK-yPN9t-qIwcI8vLRjVcYDDQklyKxKuks5smsJNgaJpZM4PZeqt>
> .
>
